src/iceberg/test/rest_catalog_test.cc

```diff
@@ -49,6 +57,70 @@ namespace iceberg::rest {
 
 namespace {
 
+// Simple local filesystem FileIO implementation for testing
+class SimpleLocalFileIO : public FileIO {
```

Comment:
I'd recommend adding `iceberg/test/std_io.h` and renaming it to `StdFileIo` to share this across tests.